src/daft-connect/src/response_builder.rs

```diff
@@ -2,34 +2,33 @@ use arrow2::io::ipc::write::StreamWriter;
 use daft_table::Table;
 use eyre::Context;
 use spark_connect::{
+    analyze_plan_response,
     execute_plan_response::{ArrowBatch, ResponseType, ResultComplete},
-    ExecutePlanResponse,
+    AnalyzePlanResponse, DataType, ExecutePlanResponse,
 };
 use uuid::Uuid;
 
-use crate::Session;
+use crate::session::Session;
 
+/// A utility for constructing responses to send back to the client,
+/// It's generic over the type of response it can build, which is determined by the type parameter `T`
+///
 /// spark responses are stateful, so we need to keep track of the session id, operation id, and server side session id
 #[derive(Clone)]
-pub struct ResponseBuilder {
+pub struct ResponseBuilder<T> {
```

Comment:
What's the point of the generic parameter here? Is it so that we can have separate methods on `ResponseBuilder<ExecutePlanResponse>` and `ResponseBuilder<AnalyzePlanResponse>`?

If that's true, then why don't we have two separate response builder classes? Not really pushing for a change, but rather more curious.

Reply:
> What's the point of the generic parameter here? Is it so that we can have separate methods on ResponseBuilder<ExecutePlanResponse> and ResponseBuilder<AnalyzePlanResponse>?

yes, im not married to the idea, but this seemed like a good solution at the time.

> If that's true, then why don't we have two separate response builder classes? Not really pushing for a change, but rather more curious.

It's because there's a bit of shared logic between the response builders, the construction is the same, but the response types are different. So my thought process was that it was a good candidate for a generic pattern like this. Happy to split it out if you think individual response builders would be more straightforward

Reply:
Actually, if there's a lot of shared logic, might as well keep it this way!